plugins/inputs/system/ps.go

```diff
@@ -73,53 +94,38 @@ func (s *SystemPS) DiskUsage(
 		return nil, nil, err
 	}
 
-	// Make a "set" out of the filter slice
-	mountPointFilterSet := make(map[string]bool)
+	mountPointFilterSet := newSet()
 	for _, filter := range mountPointFilter {
-		mountPointFilterSet[filter] = true
+		mountPointFilterSet.add(filter)
 	}
-	fstypeExcludeSet := make(map[string]bool)
+	fstypeExcludeSet := newSet()
 	for _, filter := range fstypeExclude {
-		fstypeExcludeSet[filter] = true
+		fstypeExcludeSet.add(filter)
 	}
-	paths := make(map[string]bool)
+	paths := newSet()
 	for _, part := range parts {
-		paths[part.Mountpoint] = true
+		paths.add(part.Mountpoint)
 	}
 
 	// Autofs mounts indicate a potential mount, the partition will also be
 	// listed with the actual filesystem when mounted.  Ignore the autofs
 	// partition to avoid triggering a mount.
-	fstypeExcludeSet["autofs"] = true
+	fstypeExcludeSet.add("autofs")
 
 	var usage []*disk.UsageStat
 	var partitions []*disk.PartitionStat
 	hostMountPrefix := s.OSGetenv("HOST_MOUNT_PREFIX")
 
-	for i := range parts {
-		p := parts[i]
-
-		if s.Log != nil {
-			s.Log.Debugf("[SystemPS] partition %d: %v", i, p)
-		}
-
-		if len(mountPointFilter) > 0 {
-			// If the mount point is not a member of the filter set,
-			// don't gather info on it.
-			if _, ok := mountPointFilterSet[p.Mountpoint]; !ok {
-				if s.Log != nil {
-					s.Log.Debug("[SystemPS] => dropped by mount-point filter")
-				}
-				continue
-			}
+	for _, p := range parts {
+		// If the mount point is not a member of the filter set,
+		// don't gather info on it.
+		if !mountPointFilterSet.has(p.Mountpoint) {
+			continue
 		}
```

Comment:
@reimda I think the error (detected by tests) happens here. The old code disabled filtering if there was no entry in the filter, however, your code will filter out _all_ mount-points if no `mountPointFilter` was given...

Reply:
Thanks! I hadn't noticed this